basic_features/basic_save_game_info.py

```diff
@@ -31,81 +49,166 @@ def allFiles(self) -> list[str]:
+            else:
+                pixmap = preview
+        if pixmap and not pixmap.isNull():
+            # Scale the pixmap and show it:
+            pixmap = pixmap.scaledToWidth(self._preview_width)
+            self._label.setPixmap(pixmap)
+            self._label.show()
+        else:
+            self._label.hide()
+            pixmap = None
+
+        # Add metadata, file date by default.
+        metadata = self._get_metadata(save_path, save)
+        if metadata:
+            for key, value in metadata.items():
+                self._metadata_layout.addRow(*self._new_form_row(key, str(value)))
+            self._metadata_widget.show()
+            self._metadata_widget.setLayout(self._metadata_layout)
+            self._metadata_widget.adjustSize()
+        else:
+            self._metadata_widget.hide()
 
-        # Retrieve the pixmap:
-        value = self._get_preview(Path(save.getFilepath()))
+        if metadata or pixmap:
+            self.adjustSize()
+            self.show()
 
-        if value is None:
-            return
+    def set_maximum_width(self, width: int):
+        self._preview_width = width
+        self._metadata_widget.setMaximumWidth(width)
 
-        if isinstance(value, Path):
-            pixmap = QPixmap(str(value))
-        elif isinstance(value, str):
-            pixmap = QPixmap(value)
-        elif isinstance(value, QImage):
-            pixmap = QPixmap.fromImage(value)
-        else:
-            print(
-                "Failed to retrieve the preview, bad return type: {}.".format(
-                    type(value)
-                ),
-                file=sys.stderr,
-            )
-            return
-
-        # Scale the pixmap and show it:
-        pixmap = pixmap.scaledToWidth(320)
-        self._label.setPixmap(pixmap)
-        self.resize(pixmap.width(), pixmap.height())
+    def _new_form_row(self, label: str = "", field: str = ""):
+        qLabel = QLabel(text=label)
+        qLabel.setAlignment(Qt.AlignmentFlag.AlignTop)
+        qLabel.setStyleSheet("font: italic")
+        qLabel.setSizePolicy(QSizePolicy.Policy.Minimum, QSizePolicy.Policy.Minimum)
+        qField = QLabel(text=field)
+        qField.setWordWrap(True)
+        qField.setAlignment(Qt.AlignmentFlag.AlignTop)
+        qField.setStyleSheet("font: bold")
+        qField.setSizePolicy(QSizePolicy.Policy.Preferred, QSizePolicy.Policy.Preferred)
+        return qLabel, qField
 
 
 class BasicGameSaveGameInfo(mobase.SaveGameInfo):
     def __init__(
         self,
         get_preview: Callable[[Path], QPixmap | QImage | Path | str | None]
         | None = None,
+        get_metadata: Callable[[Path, mobase.ISaveGame], Mapping[str, Any] | None]
+        | None = get_filedate_metadata,
     ):
+        """Args from: `BasicGameSaveGameInfoWidget`."""
         super().__init__()
         self._get_preview = get_preview
+        self._get_metadata = get_metadata
+        self._maximum_widget_size = 0
```

Comment:
Any reason to not pass this to the constructor and use `with_maximum_widget_size()`?